src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/AddressServiceImpl.java

```
@@ -0,0 +1,90 @@
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class AddressServiceImpl implements AddressService {
+
+    private final AddressRepository addressRepository;
+    private final UserRepository userRepository;
+    private final ModelMapper modelMapper;
+    @Override
+    public AddressResponse saveAddress(AddressRequest addressRequest, String userId) {
+        List<Address> addresses = new ArrayList<>();
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new DataNotFoundException(String.format("User with id = %s not found.", userId)));
+
+        Address address = Address.builder()
+                .city(addressRequest.getCity())
+                .street(addressRequest.getStreet())
+                .house(addressRequest.getHouse())
+                .createdAt(LocalDateTime.now())
+                .modifiedAt(LocalDateTime.now())
+                .build();
+
+        if (!addressRequest.getApartment().isEmpty()) {
+            address.setApartment(Integer.parseInt(addressRequest.getApartment()));
+        }
+        if (!addressRequest.getEntrance().isEmpty()) {
+            address.setEntrance(Integer.parseInt(addressRequest.getEntrance()));
+        }
+        if (!addressRequest.getFloor().isEmpty()) {
+            address.setFloor(Integer.parseInt(addressRequest.getFloor()));
+        }
+        if (!addressRequest.getWithLift().isEmpty()) {
+            address.setWithLift(Boolean.parseBoolean(addressRequest.getWithLift()));
+        }
+        if (user.getAddresses() != null) {
+            if (user.getAddresses().contains(address)) {
+                throw new DataAlreadyExistException(String.format("User with id %s already has this address: %s",
+                        userId, address));
```

do we really need to throw an exception if user already has inputed address?

---

if (!user.getAddresses().contains(address)) {
addresses = user.getAddresses();
}

---

yes, we throw exception so there are no duplicates in the list.

---

we need to discuss it in more detail.